File: backend/api/app/domain/repositories/message.py
from abc import ABC, abstractmethod
from app.domain.repositories import ContextManagerRepository
from core.common.either import Either
from core.errors.failure import Failure
from app.domain.entities.message import Message


class BaseWriteOnlyRepository(ContextManagerRepository):
    @abstractmethod
    async def create_message(self, message: Message, chat_id: str) -> Either[Failure, Message]:
        ...

class BaseRepository(BaseWriteOnlyRepository, ABC):
    ...

File: mobile/lib/features/architect/domains/repositories/message.dart
import 'package:architect/core/errors/failure.dart';
import 'package:architect/features/architect/domains/entities/message.dart';

import 'package:dartz/dartz.dart';

abstract class MessageRepository {
  Future<Either<Failure, Message>> createMessage({
    required String model,
    required Map<String, dynamic> payload,
    required String chatId,
  });
}
